// components/basketcart/index.js
import React, { Component } from "react";
import {Link} from 'react-router';
import { connect } from "react-redux";
import {getCartProducts} from "selectors";
import { totalmem } from "os";
export class BasketCart extends Component {
  //   static propTypes = {
  //     prop: PropTypes
  //   }
  constructor(){
      super();
      this.state = {
          cartProducts:[]
      };
  }
  componentDidMount(){
      //console.log(this.props.cartProducts)
      this.setState({cartProducts:this.props.cartProducts})
  }
    componentWillReceiveProps(nextProps){
        //console.log(nextProps)        
        this.setState({cartProducts:nextProps.cartProducts});
    }

    calculatePrice = () => {
        
        let total = 0;
        if(this.state.cartProducts){
        this.state.cartProducts.map((cartProduct)=>{
            total += cartProduct.price;
        })
    }
    console.log(total)
        return total;
    }
  render() {
      const {cartProducts} = this.state;
      //console.log(cartProducts)
      const totalPrice = this.calculatePrice();
    return (
      <div>
        <div className="cart">
          <div className="dropdown">
            <Link
              to="/basket"
              id="dLabel"
              className="btn btn-inverse btn-block btn-lg"
            >
              <i className="fa fa-fa-shopping-cart">
                <span>{`items ${cartProducts.length?cartProducts.length:0} Total ${totalPrice}`}</span>
              </i>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  cartProducts: getCartProducts(state)
});

const mapDispatchToProps = {};

export default connect(
  mapStateToProps,
  null
)(BasketCart);
